linera-core/src/updater.rs

```diff
@@ -300,15 +298,26 @@ where
                     // For `BlobsNotFound`, we assume that the local node should already be
                     // updated with the needed blobs, so sending the chain information about the
                     // certificates that last used the blobs to the validator node should be enough.
-                    let blob_ids = blob_ids
-                        .drain()
-                        .filter(|blob_id| !published_blobs.iter().any(|blob| blob.id() == *blob_id))
-                        .collect::<Vec<_>>();
-                    let missing_blob_ids = self.remote_node.node.missing_blob_ids(blob_ids).await?;
-                    let local_storage = self.local_node.storage_client();
+                    let published_blob_ids =
+                        BTreeSet::from_iter(proposal.content.block.published_blob_ids());
+                    blob_ids.retain(|blob_id| !published_blob_ids.contains(blob_id));
+                    let mut published_blobs = Vec::new();
+                    {
+                        let chain = self.local_node.chain_state_view(chain_id).await?;
```

Comment:
This change - from querying a `remote_node` to a `local_node` - seems like a win as well.

Was it even correct before?

Reply:
Ah, it was simply moved down below.